src/main/java/net/sf/rails/game/OperatingRound.java

```diff
@@ -1622,16 +1622,6 @@ public boolean layTile(LayTile action) {
             }
 
             // Sort out cost
-            /*
-            if (stl != null && stl.isFree()) {
-                cost = 0;
-            } else {
-                cost = hex.getTileCost() + extraLayTileCost(action);
-                if (stl != null) {
-                    cost = Math.max(0, cost - stl.getDiscount());
-                }
-            }
-            */
```

Comment:
Where is that check done now ? In TileLayCost() i assume ?

Reply:
No. The initial check is done in a new method gotPermission(), which is called from hexClicked(). The answer is handled in dialogActionPerformed(). All in the same class ORUIManager.

Reply:
And there you are and  i missed that comment block, sorry.